Update layer_fidelity_analysis.py to have benchmark suffix

andrehche opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/qiskit-community/qiskit-experiments?pullRequest=1583) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/qiskit-community/qiskit-experiments?pullRequest=1583) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/qiskit-community/qiskit-experiments?pullRequest=1583) it.</sub>

andrehche commented:
Thanks for the suggestions @wshanks. I modified things to use a the setter/property format to make things easier to propagate down and also added the requisite lines in the experimnet classes to use this new feature (added docstrings there since the __init__ functions there have them already. Going to test this out fully then will mark as ready for review after I'm sure it's working as intended.

wshanks commented:
Could you fix the linting errors (missing docstrings) and add a release note using `reno` (see [here](https://github.com/qiskit-community/qiskit-experiments/blob/main/CONTRIBUTING.md#adding-a-new-release-note))?

wshanks commented:
The changes look good otherwise!

andrehche commented:
Just completed the release note and marking for review! Thanks for the feedback so far.
